src/Roslyn.Diagnostics.Analyzers/UnitTests/SymbolIsBannedAnalyzerTests.cs

```diff
@@ -0,0 +1,375 @@
+            Verify(source, LanguageNames.VisualBasic, GetBasicDiagnosticAnalyzer(), additionalFiles, compilationOptions: null, parseOptions: null, expected: expected);
+        }
+
+        private void VerifyCSharp(string source, string bannedApiText, params DiagnosticResult[] expected)
+        {
+            var additionalFiles = GetAdditionalTextFiles(SymbolIsBannedAnalyzer.BannedSymbolsFileName, bannedApiText);
+            Verify(source, LanguageNames.CSharp, GetCSharpDiagnosticAnalyzer(), additionalFiles, compilationOptions: null, parseOptions: null, expected: expected);
+        }
+
+        #region Diagnostic tests
+
+        [Fact]
+        public void NoDiagnosticForNoBannedText()
+        {
+            Verify("class C { }", LanguageNames.CSharp, GetCSharpDiagnosticAnalyzer(), additionalFiles: null, compilationOptions: null, parseOptions: null, expected: Array.Empty<DiagnosticResult>());
+        }
+
+        [Fact]
+        public void NoDiagnosticReportedForEmptyBannedText()
+        {
+            var source = @"";
+
+            var bannedText = @"";
+
+            VerifyCSharp(source, bannedText);
+        }
+
+        [Fact]
+        public void DiagnosticReportedForDuplicateBannedApiLines()
+        {
+            var source = @"";
+            var bannedText = @"
+System.Console
+System.Console";
+
+            VerifyCSharp(source, bannedText,
+                GetResultAt(
+                    SymbolIsBannedAnalyzer.BannedSymbolsFileName,
+                    SymbolIsBannedAnalyzer.DuplicateBannedSymbolRule.Id,
+                    string.Format(SymbolIsBannedAnalyzer.DuplicateBannedSymbolRule.MessageFormat.ToString(), "System.Console"),
+                    "(3,1)", "(2,1)"));
+        }
+
+        [Fact]
+        public void CSharp_DiagnosticReportedForTypeInSource()
+        {
+            var source = @"
+namespace N
+{
+    class Banned { }
+    class C
+    {
+        void M()
+        {
+            var c = new Banned();
+        }
+    }
+}";
+
+            var bannedText = @"
+N.Banned";
+
+            VerifyCSharp(source, bannedText, GetCSharpResultAt(9, 21, SymbolIsBannedAnalyzer.SymbolIsBannedRule, "Banned"));
+        }
+
+        [Fact]
+        public void CSharp_DiagnosticReportedForGenericTypeFromMetadata()
+        {
+            var source = @"
+class C
+{
+    void M()
+    {
+        var c = new System.Collections.Generic.List<string>();
+    }
+}";
+
+            var bannedText = @"
+System.Collections.Generic.List`1";
+
+            VerifyCSharp(source, bannedText, GetCSharpResultAt(6, 17, SymbolIsBannedAnalyzer.SymbolIsBannedRule, "List<T>"));
+        }
+
+        [Fact]
+        public void CSharp_DiagnosticReportedForNestedType()
+        {
+            var source = @"
+class C
+{
+    class Nested { }
+    void M()
+    {
+        var n = new Nested();
+    }
+}";
+
+            var bannedText = @"
+C+Nested";
```

Comment:
Will the below test report a diagnostic?

```
            var source = @"
class C
{
    class Nested
    {
       public static void M()
       {
       }
   }
}

class D
{
   void M2()
   {
      C.Nested.M();
   }
}
";
             var bannedText = @"
C";
```

Reply:
Updated test with `Nested` as a `public static` type.

Reply:
Not currently - added some skipped tests for this locally, and will investigate.

Reply:
Fixed.